src/Solvers/plan_transforms.jl

```diff
@@ -11,7 +11,7 @@
 ##### efficient transforms. `A` will be mutated.
 #####
 
-using Oceananigans.Grids: XYRegRectilinearGrid, XZRegRectilinearGrid, YZRegRectilinearGrid, irregular_dimension
+using Oceananigans.Grids: XYRegRectilinearGrid, XZRegRectilinearGrid, YZRegRectilinearGrid, regular_dimensions, irregular_dimension
```

Comment:
We previously used "regular" and "stretched" to refer to these concepts. Have we decided to change nomenclature to "irregular" instead of "stretched"? I think it's ok to change terminology but it should be done uniformly throughtout the code rather than "by fiat" in a particular PR. The docs use "stretched grid" quite a bit so I think we would have to change this to "irregular grid"...

Reply:
Another note is that here we use "dimension" rather than "direction" as we use elsewhere. Do we want to use "dimension", or "direction"? We should pick one.

The function `irregular_dimension` is a little too specific I think, because grids can have between 0 and 3 irregular dimensions. For example, we have "regular_dimensions" --- not "regular_dimension" (the latter would only apply to a grid with a single regular dimension).

It makes sense to me to have two paired functions "regular" and "irregular" (or stretched), but they should both return _all_ of the dimensions that are either regular or stretched.

Reply:
Good points. I'm agnostic wrt stretch vs irregular grids, so I guess I'll revert back to stretched.

I'm only using `regular_dimension()` because there was already a function called just that, and I didn't wanna define a new one. However, I did try to follow the pattern where `direction` returns `XDirection` and so on, while `dimension` returns a number (3 and so on). I do think this distinction is good, but I'm open to ideas. Maybe we could use `axes`  when a number is returned and `directions` when an actual direction (e.g. `XDirection`) is returned?

And also good point about `regular_dimension`. I'll change it to plural.

Reply:
I don't mind "stretched". I also just realized that `show` uses "regularly spaced" and "variably spaced" 😂 

Anyways I think we should avoid introducing a 3rd term and probably `stretched_axes` is good. I didn't realize that `dimensions` meant it would return a number...